pfSense & OPNsense: test skipping with ENV ID

matmacieje opened the pull request:
dasharo-compatibility/os-opnsense.robot:
dasharo-compatibility/os-pfsense.robot:
Test suites are skipping in Suite Setup if BSD ENV IDs are not present in TESTED_LINUX_DISTROS variable.

[BSD_os-opnsense_2025_08_29_10_43_00_PASS.zip](https://github.com/user-attachments/files/22042007/BSD_os-opnsense_2025_08_29_10_43_00_PASS.zip)
[BSD_os-opnsense_2025_08_29_10_52_59_SKIP.zip](https://github.com/user-attachments/files/22042008/BSD_os-opnsense_2025_08_29_10_52_59_SKIP.zip)
[BSD_os-pfsense_2025_08_29_10_40_30_PASS.zip](https://github.com/user-attachments/files/22042009/BSD_os-pfsense_2025_08_29_10_40_30_PASS.zip)
[BSD_os-pfsense_2025_08_29_10_52_48_SKIP.zip](https://github.com/user-attachments/files/22042010/BSD_os-pfsense_2025_08_29_10_52_48_SKIP.zip)

DonVasyl committed:
pfSense & OPNsense: test skipping with ENV ID

dasharo-compatibility/os-opnsense.robot:
dasharo-compatibility/os-pfsense.robot:
Test suites are skipping in Suite Setup if BSD ENV IDs are not
present in TESTED_LINUX_DISTROS variable.

Signed-off-by: Mateusz Maciejewski <mateusz.maciejewski@3mdeb.com>

philipanda commented on dasharo-compatibility/os-opnsense.robot:
```diff
@@ -13,6 +13,8 @@ Resource            ../keys.robot
 Suite Setup         Run Keywords
 ...                     Prepare Test Suite
+...                     AND
+...                     Skip If    '${ENV_ID_OPNSENSE}' not in ${TESTED_LINUX_DISTROS}    OPNsense tests not supported
```
Would be good to rename this ${TESTED_LINUX_DISTROS} to some other name if its not used for Linuxes only.
I think adding OPNSENSE to TESTED_LINUX_DISTROS might cause some unexpected behavior in the BPS006 test case where it tries to run ansible on every OS from this list. Might be good to test that before merging

matmacieje replied:
Fixed in fd7853946db71e3b9e52b0fc2d42ed90a925cc90

[BSD_os-pfsense_2025_08_29_12_50_17_SKIP.zip](https://github.com/user-attachments/files/22043925/BSD_os-pfsense_2025_08_29_12_50_17_SKIP.zip)
[BSD_os-pfsense_2025_08_29_12_32_18_PASS.zip](https://github.com/user-attachments/files/22043926/BSD_os-pfsense_2025_08_29_12_32_18_PASS.zip)
[BSD_os-opnsense_2025_08_29_12_50_05_SKIP.zip](https://github.com/user-attachments/files/22043927/BSD_os-opnsense_2025_08_29_12_50_05_SKIP.zip)
[BSD_os-opnsense_2025_08_29_12_35_08_PASS.zip](https://github.com/user-attachments/files/22043928/BSD_os-opnsense_2025_08_29_12_35_08_PASS.zip)

DonVasyl committed:
TESTED_BSD_DISTROS variable introduced to default.robot

New "tested distros" variable introduced to skip BSD tests (pfSense & OPN
sense). Empty list defined in default.robot.

Signed-off-by: Mateusz Maciejewski <mateusz.maciejewski@3mdeb.com>

macpijan commented on platform-configs/include/default.robot:
```diff
@@ -294,6 +294,7 @@ ${DCU_SUPPORTED_BOOLEAN_SMMSTORE_VARIABLE}=         NetworkBoot
 ${WINDOWS_SHUTDOWN_AWAITING_SECONDS}=               120
 
 @{TESTED_LINUX_DISTROS}=                            ${ENV_ID_UBUNTU}
+@{TESTED_BSD_DISTROS}=                              @{EMPTY}
```
Should this be enabled for any of the platforms?

philipanda replied:
I think that can be done when there will be a need to test them?

matmacieje replied:
@macpijan Moved both test suites from dasharo-compatibility to dasharo-oses, TESTED_BSD_DISTROS populated for all Protectli platforms. Changes in 3caac454a52359d84859b141640042588cd31f93

macpijan replied:
ok let's try this approach

philipanda commented on platform-configs/include/default.robot:
```diff
@@ -294,6 +294,7 @@ ${DCU_SUPPORTED_BOOLEAN_SMMSTORE_VARIABLE}=         NetworkBoot
 ${WINDOWS_SHUTDOWN_AWAITING_SECONDS}=               120
 
 @{TESTED_LINUX_DISTROS}=                            ${ENV_ID_UBUNTU}
+@{TESTED_BSD_DISTROS}=                              @{EMPTY}
```
What about doing something like this:
```robotframework
@{TESTED_LINUX_DISTROS}=     ${ENV_ID_UBUNTU}
@{TESTED_BSD_DISTROS}=    ${ENV_ID_OPNSENSE}
@{TESTED_OSES}=    @{TESTED_LINUX_DISTROS}    @{TESTED_BSD_DISTROS}    ${ENV_ID_WINDOWS}    ${ENV_ID_SOMETHING}
```

Using `@{variable}` in a definitino should explode it into its items effectively joining all these lists. I think that none of the current usages of `TESTED_LINUX_DISTROS` uses the fact that they are actually Linux distros, so it could be replaced with this `TESTED_OSES`. If someone would need to get a list of Linux or BSD distros for compatibilty, then that info would be available for them too, but the problem of checking what OS is available to be booted would be more intuitive I think.

matmacieje replied:
I can add TESTED_OSES but I'm not going to put it everywhere in OSFV in scope of this task.

philipanda replied:
It's a search&replace, but it might potentially cause some problems so might be better to keep it as little intrusive as possible, so it's fine to moe

philipanda commented on test_cases.json:
```diff
@@ -5058,144 +5058,144 @@
       "_id": "OPN001.001",
       "name": "OPNsense stable (serial output) installation on Hard Disk",
-      "module": "Dasharo Compatibility",
+      "module": "Dasharo OSes",
```
Introducing new test modules will break a couple things related to, but not only, osfv-dashboard.
- the test ID validity checks in CI at [`scripts/ci/test_ids/test_ids_legal.py`](https://github.com/Dasharo/open-source-firmware-validation/blob/develop/scripts/ci/test_ids/test_ids_legal.py#L13)
- pre-commit matches every directory that starts with "dasharo*", so its fine: https://github.com/Dasharo/open-source-firmware-validation/blob/develop/.pre-commit-config.yaml#L121
- README will break too: https://github.com/Dasharo/open-source-firmware-validation/blob/develop/README.md?plain=1#L69
- (...more?)

Please search for any instances of the test module names and fix them if this change is to stay

matmacieje replied:
Change reverted, os-pfsense.robot & os-opnsense.robot now moved to dasharo-compatibility.

+ rebase.

DonVasyl committed:
TESTED_BSD_DISTROS populated for Protectli platforms.

Signed-off-by: Mateusz Maciejewski <mateusz.maciejewski@3mdeb.com>

matmacieje commented:
Rebase.
